src/librbd/io/AioCompletion.h

```diff
@@ -38,21 +38,15 @@ namespace io {
  * context or via a thread pool context for cache read hits).
  */
 struct AioCompletion {
-  typedef enum {
-    AIO_STATE_PENDING = 0,
-    AIO_STATE_CALLBACK,
-    AIO_STATE_COMPLETE,
-  } aio_state_t;
-
   mutable std::mutex lock;
-  std::condition_variable cond;
+  std::atomic<bool> waiting_for_complete{false};
 
   callback_t complete_cb = nullptr;
   void *complete_arg = nullptr;
   rbd_completion_t rbd_comp = nullptr;
 
   /// note: only using atomic for built-in memory barrier
-  std::atomic<aio_state_t> state{AIO_STATE_PENDING};
```

Comment:
The comment above is no longer true -- we have grown wait/notify use.  Let's remove it.